selfservice/hook/web_hook_integration_test.go

```diff
@@ -607,6 +609,90 @@ func TestWebHooks(t *testing.T) {
+
+		t.Run("case=update identity fields", func(t *testing.T) {
+			expected := identity.Identity{
+				Credentials: map[identity.CredentialsType]identity.Credentials{identity.CredentialsTypePassword: {Type: "password", Identifiers: []string{"test"}, Config: []byte(`{"hashed_password":"$argon2id$v=19$m=65536,t=1,p=1$Z3JlZW5hbmRlcnNlY3JldA$Z3JlZW5hbmRlcnNlY3JldA"}`)}},
+				SchemaID:    "default",
+				SchemaURL:   "file://stub/default.schema.json",
+				State:       identity.StateActive,
+				Traits:      []byte(`{"email":"some@example.org"}`),
+				VerifiableAddresses: []identity.VerifiableAddress{{
+					Value:    "some@example.org",
+					Verified: false,
+					Via:      "email",
+					Status:   identity.VerifiableAddressStatusPending,
+				}},
+				RecoveryAddresses: []identity.RecoveryAddress{{
+					Value: "some@example.org",
+					Via:   "email",
+				}},
+				MetadataPublic:      []byte(`{"public":"data"}`),
+				MetadataAdmin:       []byte(`{"admin":"data"}`),
+				InternalCredentials: identity.CredentialsCollection{{Type: "password", Identifiers: []string{"test"}, Config: []byte(`{}`)}},
+			}
+
+			t.Run("case=body is empty", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity is present but empty", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated traits", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"traits":{"email":"some@other-example.org"}}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated state", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"state":"inactive"}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated public metadata", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"metadata_public":{"useful":"metadata"}}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated admin metadata", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"metadata_admin":{"useful":"metadata"}}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated verified addresses", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"traits":{"email":"some@other-example.org"},"verifiable_addresses":[{"value":"some@other-example.org","via":"email"}]}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
+
+			t.Run("case=identity has updated recovery addresses", func(t *testing.T) {
+				actual := run(t, expected, http.StatusOK, []byte(`{"identity":{"traits":{"email":"some@other-example.org"},"recovery_addresses":[{"value":"some@other-example.org","via":"email"}]}}`))
+				snapshotx.SnapshotT(t, &actual)
+			})
```

Comment:
Good point, I'm pretty sure that this would latest break during the registration itself, but validating that the JSON is valid can probably be done prior with an improved error message

Reply:
I think this is sufficiently handled in the manager itself, that is later being called. Generally speaking, webhook observability needs to be improved in any case.